SteamKit2/SteamKit2/Steam/Handlers/SteamAuthTicket/DefaultAuthTicketBuilder.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Diagnostics;
+using System.IO;
+
+namespace SteamKit2
+{
+    /// <summary>
+    /// Handles serialization of user specific data in auth token generation process
+    /// </summary>
+    public interface IUserDataSerializer
+    {
+        /// <summary>
+        /// Writes user data into provided buffer.
+        /// </summary>
+        /// <param name="writer">Buffer to write data into.</param>
+        void Serialize( BinaryWriter writer );
+    }
+
+    /// <summary>
+    /// Handles building auth tickets.
+    /// </summary>
+    public interface IAuthTicketBuilder
```

Comment:
I don't really understand why these interfaces exist, you can only implement whatever format Steam does, and consumes aren't going to be providing their own, can all of this be simplified to be less OOP?